layer/clients/model_catalog.py

```diff
@@ -218,64 +175,54 @@ def _load_model_runtime_objects(
 
     def save_model_object(
         self,
-        model: Model,
         model_object: ModelObject,
+        train_id: uuid.UUID,
         transfer_state: ResourceTransferState,
     ) -> ModelObject:
-        self._logger.debug(f"Storing given model {model_object} for {model.path}")
+        flavor = get_flavor_for_model(model_object)
+        if flavor is None:
+            raise UnexpectedModelTypeException(type(model_object))
```

Comment:
somewhat unrelated, but probably you wanna improve the message `UnexpectedModelTypeException` throws.

Reply:
we will redo artifacts soon, let's tackle that then?